### PR TITLE
Add hover behavior for dnd items

### PR DESCRIPTION
**Pull request by kaitoo1**
This PR adds the following:
      - onHover, show remove button that unstages item when clicked
      - onHover, show nft details label
      - added state for all nfts (displayed in sidebar) so dnd can access whether item is selected
      - refactored structure of dnd and sidebar

Below is an overview of the new component structure and naming for the AddNfts step.

- `AddNfts` (step)
  - `CollectionEditor`
    - `Sidebar`
      - `SidebarNftIcon` (formerly called `NftPreviewIcon`)
    - `StagingArea` (for dnd, formerly called `Editor`)
      - `StagedNftWrapper` (renders a staged nft in dnd. wraps the sortable nft and unstage button)
        - `UnstageButton` (unstages nft on click)
        - `SortableStagedNft` (dnd wrapper element that makes nft sortable)
          - `StagedNftImage` (renders nft image + details label)

### CODE REVIEW

File: src/scenes/CollectionCreationFlow/steps/AddNfts/Editor/StagedNftWrapper.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  nft: Nft;
+  activeId?: string;
+};
+
+function StagedNftWrapper({ nft, activeId }: Props) {
+  return (
+    <StyledStageNftWrapper>
+      <StyledUnstageButton
+        // TODO make index guaranteed
+        nftIndex={nft.index || 0}
```
Comment (kaitoo1) [inline src/scenes/CollectionCreationFlow/steps/AddNfts/Editor/StagedNftWrapper.tsx:17]: I think we should differentiate between the existing `Nft` type and an "editable" `Nft` type that has fields specific to the context of the Collection Editor, like `index` and `isSelected`

Comment (kaitoo1) [reply src/scenes/CollectionCreationFlow/steps/AddNfts/Editor/StagedNftWrapper.tsx:17]: or wrap the `Nft` type in a new object that looks like 
{
nft: Nft
index: number
isSelected: boolean
}
that's probably better

Comment (kaitoo1) [reply src/scenes/CollectionCreationFlow/steps/AddNfts/Editor/StagedNftWrapper.tsx:17]: what do you think @Robinnnnn

File: src/contexts/collectionEditor/CollectionEditorContext.tsx [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+  return pics;
+}
+
+export type AllNftsState = EditModeNft[];
+export type StagedNftsState = EditModeNft[];
+
+export type CollectionEditorState = {
+  allNfts: EditModeNft[];
+  stagedNfts: EditModeNft[];
+};
```
Comment (Robinnnnn) [inline src/contexts/collectionEditor/CollectionEditorContext.tsx:48]: should use these states in CollectionEditorState

File: src/scenes/CollectionCreationFlow/steps/AddNfts/Editor/StagedNftImageDragging.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  const handleMouseUp = useCallback(() => {
+    setIsMouseUp(true);
+  }, []);
+
+  useEffect(() => {
+    document.addEventListener('mouseup', handleMouseUp);
+    return () => {
+      document.removeEventListener('mouseup', handleMouseUp);
+    };
+  }, [handleMouseUp]);
```
Comment (Robinnnnn) [inline src/scenes/CollectionCreationFlow/steps/AddNfts/Editor/StagedNftImageDragging.tsx:21]: can be converted to a useMouseX hook

File: src/scenes/CollectionCreationFlow/steps/AddNfts/Editor/StagedNftWrapper.tsx (hunk 1 above)
Comment (Robinnnnn) [reply src/scenes/CollectionCreationFlow/steps/AddNfts/Editor/StagedNftWrapper.tsx:17]: makes total sense!

File: src/scenes/CollectionCreationFlow/steps/AddNfts/Sidebar/SidebarNftIcon.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
+import { useCollectionEditorActions } from 'contexts/collectionEditor/CollectionEditorContext';
+import useEffectAfterMount from 'hooks/useEffectAfterMount';
+import { memo, useCallback, useMemo } from 'react';
+import styled from 'styled-components';
+import { EditModeNft } from 'types/Nft';
+
+type SidebarNftIconProps = {
+  nft: EditModeNft; // TODO: this will be an object in the future
```
Comment (Robinnnnn) [inline src/scenes/CollectionCreationFlow/steps/AddNfts/Sidebar/SidebarNftIcon.tsx:8]: toss TODO

File: src/scenes/CollectionCreationFlow/steps/AddNfts/Sidebar/SidebarNftIcon.tsx [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    stageNft,
+    unstageNft,
+  } = useCollectionEditorActions();
+
+  const handleClick = useCallback(() => {
+    setNftIsSelected(index, !isSelected);
+  }, [index, isSelected, setNftIsSelected]);
+
+  const handleUpdate = useCallback(() => {
+    isSelected ? stageNft(nft) : unstageNft(nft.nft.id);
```
Comment (Robinnnnn) [inline src/scenes/CollectionCreationFlow/steps/AddNfts/Sidebar/SidebarNftIcon.tsx:25]: nft.nft.nft.nft

**Comment by Robinnnnn** [comment]
UI comment: there should be a shadow on the top/bottom, and the bottom shadow should be a bit lighter/shorter

image

**Review by Robinnnnn — APPROVED** [review]

File: src/contexts/collectionEditor/CollectionEditorContext.tsx (hunk 2 above)
Comment (kaitoo1) [reply src/contexts/collectionEditor/CollectionEditorContext.tsx]: done

File: src/scenes/CollectionCreationFlow/steps/AddNfts/Editor/StagedNftImageDragging.tsx (hunk 3 above)
Comment (kaitoo1) [reply src/scenes/CollectionCreationFlow/steps/AddNfts/Editor/StagedNftImageDragging.tsx]: done, added useMouseUp hook

File: src/scenes/CollectionCreationFlow/steps/AddNfts/Sidebar/SidebarNftIcon.tsx (hunk 4 above)
Comment (kaitoo1) [reply src/scenes/CollectionCreationFlow/steps/AddNfts/Sidebar/SidebarNftIcon.tsx]: done

File: src/scenes/CollectionCreationFlow/steps/AddNfts/Sidebar/SidebarNftIcon.tsx (hunk 5 above)
Comment (kaitoo1) [reply src/scenes/CollectionCreationFlow/steps/AddNfts/Sidebar/SidebarNftIcon.tsx]: lolol renamed prop to editModeNft

**Pull request merged**